File: app/src/main/java/com/example/polyquicktrade/ui/ask/AskAdapter.java
package com.example.polyquicktrade.ui.ask;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.polyquicktrade.pojo.Enquiry;
import com.example.polyquicktrade.R;
import com.example.polyquicktrade.RecyclerClickListener;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

class AskAdapter extends RecyclerView.Adapter<AskAdapter.AskViewHolder> {


    private final List<Enquiry> enquiries;
    private final RecyclerClickListener<Enquiry> clickListener;

    public static class AskViewHolder extends RecyclerView.ViewHolder{

        TextView enquiryNameTV, enquireDescTV;
        CardView cardView;

        public AskViewHolder(@NonNull View itemView) {
            super(itemView);

            enquiryNameTV = itemView.findViewById(R.id.EnquiryNameTextView);
            enquireDescTV = itemView.findViewById(R.id.EnquiryDescTextView);
            cardView = itemView.findViewById(R.id.EnquiryCardView);
        }

        public void bindListener(final Enquiry enquiry, final RecyclerClickListener<Enquiry> clickListener) {

            cardView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    clickListener.onClick(enquiry, view);
                }
            });
        }
    }

    public AskAdapter(List<Enquiry> enquiries, RecyclerClickListener<Enquiry> clickListener) {

        this.enquiries = enquiries;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public AskViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.custom_ask_view, parent, false);

        return new AskViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull AskViewHolder holder, int position) {

        final Enquiry enquiry = this.enquiries.get(position);

        holder.enquiryNameTV.setText(enquiry.getProductName());
        holder.enquireDescTV.setText(enquiry.getProductDesc());

        holder.bindListener(enquiry, this.clickListener);

    }

    @Override
    public int getItemCount() {
        return enquiries.size();
    }
}

File: app/src/main/java/com/example/polyquicktrade/ui/ask/AskAdapter.kt
package com.example.polyquicktrade.ui.ask

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.example.polyquicktrade.R
import com.example.polyquicktrade.RecyclerClickListener
import com.example.polyquicktrade.pojo.Enquiry
import com.example.polyquicktrade.ui.ask.AskAdapter.AskViewHolder

internal class AskAdapter(
    private val enquiries: List<Enquiry>,
    private val clickListener: RecyclerClickListener<Enquiry?>
) : RecyclerView.Adapter<AskViewHolder>() {
    class AskViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var enquiryNameTV: TextView = itemView.findViewById(R.id.EnquiryNameTextView)
        var enquireDescTV: TextView
        var cardView: CardView
        fun bindListener(enquiry: Enquiry?, clickListener: RecyclerClickListener<Enquiry?>) {
            cardView.setOnClickListener { view -> clickListener.onClick(enquiry, view) }
        }

        init {
            enquireDescTV = itemView.findViewById(R.id.EnquiryDescTextView)
            cardView = itemView.findViewById(R.id.EnquiryCardView)
        }
    }
//
//    fun add(newEnquiries: List<Enquiry>){
//        enquiries.
//    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AskViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val v = inflater.inflate(R.layout.custom_ask_view, parent, false)
        return AskViewHolder(v)
    }

    override fun onBindViewHolder(holder: AskViewHolder, position: Int) {
        val enquiry = enquiries[position]
        holder.enquiryNameTV.text = enquiry.productName
        holder.enquireDescTV.text = enquiry.productDesc
        holder.bindListener(enquiry, clickListener)
    }

    override fun getItemCount(): Int {
        return enquiries.size
    }


}